### PR TITLE
Add state definitions to fix tests

### PR DESCRIPTION
**Pull request by sheam**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by skirsdeda** [comment]
👎 Please provide some description of a problem you're trying to solve with this. I don't think React.Component type definition changed...

**Comment by skirsdeda** [comment]
@sheam Sorry, overlooked that no state was defined. But you should try defining the actual state object instead of `any`: https://github.com/JedWatson/react-select/blob/master/src/Select.js#L69. And version change is not needed (must correspond to react-select version).

**Comment by sheam** [comment]
Ok, I will add state. I thought the fact that I was getting it to a usable state was improvement enough. :)

Give me a couple minutes here.

**Comment by sheam** [comment]
Ok, I have updated to add a state.

**Comment by skirsdeda** [comment]
@sheam Actually `Select.Async` has different state while `Creatable` and `AsyncCreatable` are stateless. Look at the code :)

**Comment by sheam** [comment]
Ok, I added an async state, and used undefined for the Creatable versions.

**Review by skirsdeda — CHANGES_REQUESTED** [review]

File: types/react-select/index.d.ts [hunk 1]
```diff
@@ -485,7 +485,7 @@ declare namespace ReactSelectClass {
 declare class ReactSelectClass extends React.Component<ReactSelectClass.ReactSelectProps, ReactSelectClass.State> { }
 
 declare module ReactSelectClass {
-    class Creatable extends React.Component<ReactCreatableSelectProps, undefined> { }
+    class Creatable extends React.Component<ReactCreatableSelectProps, {}> { }
     class Async extends React.Component<ReactAsyncSelectProps, ReactSelectClass.AsyncState> { }
-    class AsyncCreatable extends React.Component<ReactAsyncSelectProps & ReactCreatableSelectProps, undefined> { }
+    class AsyncCreatable extends React.Component<ReactAsyncSelectProps & ReactCreatableSelectProps, {}> { }
```
Comment (skirsdeda) [inline types/react-select/index.d.ts]: Tests are failing. I guess this should have `ReactSelectClass.AsyncState` as it's state.

Comment (sheam) [inline types/react-select/index.d.ts]: Yesh, sorry just noticed that now. Just pushed fix.

**Comment by skirsdeda** [comment]
@sheam Looking good now but another big PR just got merged (#18287) so you'll have to rebase it. `Option[]` now becomes `Options`. Everything else should be good I think.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has failing CI tests and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**